template/interpolate/azure/keyvault/keyvault.go

```diff
@@ -0,0 +1,216 @@
+package keyvault
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	keyvaultmgmt "github.com/Azure/azure-sdk-for-go/services/keyvault/2016-10-01/keyvault"
+	"github.com/Azure/azure-sdk-for-go/services/keyvault/mgmt/2018-02-14/keyvault"
+	"github.com/hashicorp/go-azure-helpers/authentication"
+	"github.com/hashicorp/go-azure-helpers/sender"
+)
+
+// ResourceID represents a parsed long-form Azure Resource Manager ID
+// with the Subscription ID, Resource Group and the Provider as top-
+// level fields, and other key-value pairs available via a map in the
+// Path field.
+type ResourceID struct {
+	SubscriptionID string
+	ResourceGroup  string
+	Provider       string
+	Path           map[string]string
+}
+
+type Client struct {
+	VaultsClient     *keyvault.VaultsClient
+	ManagementClient *keyvaultmgmt.BaseClient
+}
+
+func New() (*Client, error) {
+	c := &Client{}
+
+	vaultsClient, managementClient, err := newVaultsClients()
+	if err != nil {
+		return nil, err
+	}
+
+	c.VaultsClient = vaultsClient
+	c.ManagementClient = managementClient
+
+	return c, nil
+}
+
+func (c *Client) GetKeyVaultIdFromRgAndName(rg string, kvName string) (string, error) {
+	resp, err := c.VaultsClient.Get(context.Background(), rg, kvName)
+	if err != nil {
+		return "", err
+	}
+
+	return *resp.ID, nil
+}
+
+func (c *Client) GetSecretFromKeyVaultIdAndKey(keyVaultId string, key string) (string, error) {
```

Comment:
```suggestion
func (c *Client) GetSecretFromKeyVault(keyVaultId string, key string) (string, error) {
```

The parameter names help give context to the method I think we can shorten the name a bit and document the method.

Reply:
function names changed